cap/csp/server/executor.go

```diff
@@ -68,16 +75,46 @@ func (r Runtime) mkmod(ctx context.Context, args api.Executor_exec_Params) (wasm
 
 	// TODO(perf):  cache compiled modules so that we can instantiate module
 	//              instances for concurrent use.
-	module, err := r.Runtime.CompileModule(ctx, bc)
+	compiled, err := r.Runtime.CompileModule(ctx, bc)
 	if err != nil {
 		return nil, err
 	}
 
-	return r.Runtime.InstantiateModule(ctx, module, wazero.
-		NewModuleConfig().
+	// TODO(perf): find a way of locating a free port without opening and
+	//             closing a connection.
+	// Find a free TCP port.
+	l, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, err
+	}
+	addr := l.Addr().(*net.TCPAddr)
+
+	// Enables the creation of non-blocking TCP connections
+	// inside the WASM module. The host will pre-open the TCP
+	// port and pass it to the guest through a file descriptor.
+	sockCfg := sock.NewConfig().WithTCPListener("", addr.Port)
+	sockCtx := sock.WithConfig(ctx, sockCfg)
+	modCfg := wazero.NewModuleConfig().
+		WithStartFunctions(). // don't call _start until later
+		WithSysNanosleep().
+		WithSysNanotime().
+		WithSysWalltime().
+		WithRandSource(rand.Reader).
 		WithName(name).
-		WithStartFunctions(). // disable automatic calling of _start (main)
-		WithRandSource(rand.Reader))
+		WithEnv("ns", name).
+		WithStdin(os.Stdin).
+		WithStdout(os.Stdout).
+		WithStderr(os.Stderr)
+
+	l.Close()
```

Comment:
Should this be deferred?

Reply:
I don't think so! The port is now in use and we need it freed so it can be pre-opened for the process.

It was opened before because it's the way to ensure the port is free when we pre-open it. If you know any other way we can change it

Reply:
Ah, I see!  Never mind!